File: 프로그래머스/3/43162. 네트워크/네트워크.java
import java.util.*;

class Solution {
    static int[] visited;
        
    static int cnt = 0;
    
    public int solution(int n, int[][] computers) {
        int answer = 0;
        
        visited = new int[n];
        
        for(int i = 0; i < n; i++) {
            if(visited[i] == 1) continue;
            visited[i] = 1;
            cnt++;
            bfs(n, computers, i);
        }
        
        answer = cnt;
        
        return answer;
    }
    
    public static void bfs(int n, int[][] computers, int idx) {        
        Queue<Integer> q = new LinkedList<>();
        
        q.add(idx);
        
        while(q.isEmpty() == false) {
            Integer pos = q.poll();
            
            for(int i = 0; i < n; i++) {
                if(pos == i) continue;
                if(computers[pos][i] == 0) continue;
                if(visited[i] == 1) continue;
                
                visited[i] = 1;
                q.add(i);
            }
            
        }
    }
}

File: 프로그래머스/3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    var answer = 0;
    
    let vis = []
    
    const bfs = (idx, vis) => {
        if(vis.includes(idx)) return
        
        // console.log("bfs(", idx, ")")
        
        let q = []
        vis.push(idx)
        q.push(idx)
        
        answer++

        while(q.length !== 0) {
            let ci = q.shift()
            // console.log('cur ci=', ci)

            for(let i = 0; i < n; i++) {
                if(vis.includes(i)) continue
                if(computers[ci][i] !== 1) continue
                if(ci === i) continue

                vis.push(i)
                q.push(i)
            }
        }
    }

    for(let i = 0; i < n; i++) {
        bfs(i, vis)        
    }
    
    return answer;
}